website/static/js/registrationUtils.js

```diff
@@ -396,8 +403,7 @@ Draft.prototype.register = function(data) {
         .done(function(response) {
```

Comment:
All methods that do HTTP work should return a promise object. That makes it easier to bind additional actions to the success/failure of the request, outside of the method.

(eg if your ViewModel wanted to do some additional error handling/ set observables after the completion of a request)

See COSDev docs for a basic rundown, plus some more complicated examples:
https://cosdev.readthedocs.org/en/latest/style_guides/javascript.html#promises